### PR TITLE
Add support for multiple "rcpt to" users.

### PR DESCRIPTION
**Pull request by rfk**
Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1362218#c1; @vbudhram r?; /cc @jrgm

### CODE REVIEW

File: emailserver.js [hunk 1]
```diff
@@ -45,21 +45,23 @@ var server = smtp.createServer(HOSTNAME, function (req) {
 
     mailparser.on('end', function(mail) {
       mail.receivedAt = new Date().toISOString();
-      var user = req.to;
-      log('Received message for', user);
+      var users = req.to;
+      log('Received message for', users);
```
Comment (rfk) [inline emailserver.js]: It turns out that `req.to` is always an array, even if there's only a single user.  So I'm not sure why this was working properly before, maybe some magic in the redis module?

Comment (jrgm) [inline emailserver.js]: This rings a bell. I have a refresh branch to move this to node4 (from about a year ago) and I had to fix this problem. Must follow through with that branch (which includes a load test).

**Comment by jrgm** [comment]
iirc, that was this branch https://github.com/mozilla/restmail.net/commits/ansible-cloudformation-deploy

**Comment by jrgm** [comment]
I think this - https://github.com/mozilla/restmail.net/commit/413e47e38b6eeb047b69cba64b02381f7fd8ea87#diff-9ecbdcf51b33c9b223c2d7b7c899fdd8L48

File: emailserver.js (hunk 1 above)
Comment (vbudhram) [inline emailserver.js]: Would this change also make it work for `bcc` emails?

Comment (rfk) [inline emailserver.js]: I believe so, yes.

**Comment by rfk** [comment]
@jrgm I pulled in the dependency updates, and bumped the node version because it wasn't passing on travis with node 0.10.  Could you please give this an r? and if it looks OK from your perspective, merge and release?

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 language: node_js
 
 node_js:
- - "0.10"
+ - "4.5"
```
Comment (jrgm) [inline .travis.yml]: Should really just say "4", since current is 4.8.x

**Comment by jrgm** [comment]
Looking at detail log on travis-ci, the hiredis dependency doesn't build, but it's optional
https://travis-ci.org/mozilla/restmail.net/builds/230995200#L166

At some point should clean that up (and perhaps drop that optional dep, since it may be obsolete).

**Comment by rfk** [comment]
> Looking at detail log on travis-ci, the hiredis dependency doesn't build

Interesting, looks like maybe it's just because we're missing the c++ buildchain?

**Comment by jrgm** [comment]
I vaguely recall looking at the difference between having hiredis and not having it in the loadtest, and there was no real difference. So we can just remove it for our use case.

**Pull request merged**